src/Controller/ProfileController.php

```diff
@@ -15,13 +19,42 @@ class ProfileController extends AbstractController
 {
     /**
      * @Route("/{username}", name="show")
-     * @ParamConverter("registeredUser", class="App\Entity\RegisteredUser",
+     * @ParamConverter("user", class="App\Entity\User"),
      * options={"mapping": {"username": "username"}})
      */
-    public function show(RegisteredUser $registeredUser): Response
+    public function show(User $user): Response
     {
         return $this->render('profile/show.html.twig', [
-            'registeredUser' => $registeredUser,
+            'user' => $user,
+        ]);
+    }
+
+    /**
+     * @Route("/{username}/edit", name="edit", methods={"GET","POST"})
+     * @ParamConverter("user", class="App\Entity\User"),
+     * options={"mapping": {"username": "username"}})
+     */
+    public function edit(Request $request, User $user, EntityManagerInterface $entityManager): Response
+    {
+        $registeredUser = new RegisteredUser();
+        $registeredUser->setUser($user);
+
+        $form = $this->createForm(RegisteredUserType::class, $user->getRegisteredUser() ?? $registeredUser);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            if (!$user->getRegisteredUser()) {
+                $entityManager->persist($registeredUser);
+            }
+
+            $entityManager->flush();
+
+            return $this->redirectToRoute('profile_show', ['username' => $user->getUsername()]);
+        }
+
+        return $this->render('profile/edit.html.twig', [
+            'user' => $user,
+            'form' => $form->createView(),
```

Comment:
so anyone can edit anyone's profile? shouldn't you redirect or send a 403 code right away if someone tries to edit someone else's profile